java-connectors/kafka-connect-azure-servicebus/src/main/java/io/lenses/streamreactor/connect/azure/servicebus/source/AzureServiceBusSourceConnector.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2017-2024 Lenses.io Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.lenses.streamreactor.connect.azure.servicebus.source;
+
+import static io.lenses.streamreactor.common.util.AsciiArtPrinter.printAsciiHeader;
+
+import io.lenses.streamreactor.common.util.JarManifest;
+import io.lenses.streamreactor.connect.azure.servicebus.config.AzureServiceBusConfigConstants;
+import io.lenses.streamreactor.connect.azure.servicebus.config.AzureServiceBusSourceConfig;
+import io.lenses.streamreactor.connect.azure.servicebus.util.KcqlConfigBusMapper;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.IntStream;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.Task;
+import org.apache.kafka.connect.source.SourceConnector;
+
+/**
+ * Implementation of {@link SourceConnector} for Microsoft Azure EventHubs.
+ */
+@Slf4j
+public class AzureServiceBusSourceConnector extends SourceConnector {
+
+  private final JarManifest jarManifest =
+      new JarManifest(getClass().getProtectionDomain().getCodeSource().getLocation());
+  private Map<String, String> configProperties;
+
+  @Override
+  public void start(Map<String, String> props) {
+    parseAndValidateConfigs(props);
+    configProperties = props;
+    printAsciiHeader(jarManifest, "/azure-servicebus-ascii.txt");
+  }
+
+  private void parseAndValidateConfigs(Map<String, String> props) {
+    new AzureServiceBusSourceConfig(props);
+    KcqlConfigBusMapper.mapKcqlsFromConfig(props.get(AzureServiceBusConfigConstants.KCQL_CONFIG));
```

Comment:
Here we're relying on the fact that these two methods will throw exceptions if the configurations supplied are not correct.  It's easy to miss this reading the code.

How can we make this more obvious?

Wouldn't making sure each of these throw a specific checked exception that we would be forced to handle be the Java way of signalling this?

Reply:
Debate about Exceptions and whether to handle them or not is going on for years now, even Oracle having [their stance](https://docs.oracle.com/javase/tutorial/essential/exceptions/runtime.html) and autority figures like Robert Martin having [different ones](https://santim0ren0.medium.com/clean-code-error-handling-2f973908b13e).  (_I tend to be more with Martin on it especially after introducing streams to Java, but it's not place nor the time_)

This way or another, this is a classic scenario of unrecoverable situation - Connector is having bad configuration passed to it, we want to fail fast. That's why I think there's no sense in using checked exception, nor create all the boilerplate (`try-catch` and `throws` for all the methods up to the Connector itself).

Reply:
Absolutely valid sources.  Thank you for sharing.  But as said previously we wish to consider a more functional approach to our code that reduces side-effects and makes the flow clearer staying away from unintended consequences.  We've taken this discussion into the team channel and will update when we have a resolution. 👍

Reply:
I'm going to resolve this for now and will raise a technical debt ticket to address when we have decided functional error handling approach.